cmd/babble/commands/root.go

```diff
@@ -0,0 +1,98 @@
+package commands
+
+import (
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	config = NewDefaultCLIConfig()
+	logger = logrus.New()
+)
+
+//RootCmd is the root command for Babble
+var RootCmd = &cobra.Command{
+	Use:              "babble",
+	Short:            "babble consensus",
+	TraverseChildren: true,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) (err error) {
+		if cmd.Name() == VersionCmd.Name() {
+			return nil
+		}
+
+		if err := bindFlagsLoadViper(cmd); err != nil {
+			return err
+		}
+
+		config, err = parseConfig()
+		if err != nil {
+			return err
+		}
+
+		logger.Level = logLevel(config.Babble.LogLevel)
+
+		logger.WithFields(logrus.Fields{
+			"datadir": config.Babble.DataDir,
+			"log":     config.Babble.LogLevel,
+		}).Debug("Config")
+
+		return nil
+	},
+}
+
+//------------------------------------------------------------------------------
+
+//Bind all flags and read the config into viper
+func bindFlagsLoadViper(cmd *cobra.Command) error {
+	// cmd.Flags() includes flags from this command and all persistent flags from the parent
+	if err := viper.BindPFlags(cmd.Flags()); err != nil {
+		return err
+	}
+
+	viper.SetConfigName("config")                                       // name of config file (without extension)
+	viper.AddConfigPath(config.Babble.DataDir)                          // search root directory
+	viper.AddConfigPath(filepath.Join(config.Babble.DataDir, "config")) // search root directory /config
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+		// stderr, so if we redirect output to json file, this doesn't appear
+		// fmt.Fprintln(os.Stderr, "Using config file:", viper.ConfigFileUsed())
+	} else if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+		// ignore not found error, return other errors
```

Comment:
We may want to warn the user that no config file is found, and that we took the default values (useful for the user to know that he can have a config file, and if he intended to have one, to know that it is not found)